### PR TITLE
Introduce Component as the base type for a few components

### LINKED ISSUE #12154: Add ComponentModel base classes where missing.
All these should derive from Component.

- [x] System.Timers.Timer
- [x] System.Net.WebClient
- [x] System.Net.NetworkInformation.Ping
- [x] System.ComponentModel.BackgroundWorker
- [x] System.IO.FileSystemWatcher
- [x] System.Diagnostics.Process
- [x] System.Diagnostics.ProcessModule
- [x] System.Diagnostics.ProcessThread

### PR DESCRIPTION
**Pull request by AlexGhiondea**
The following types were changed to derive from Component:
- WebClient
- FileSystemWatcher
- Ping
- BackgroundWorker
- Process
- ProcessModule
- ProcessThread

Fixes #12154

### CODE REVIEW

File: src/System.ComponentModel.EventBasedAsync/src/System/ComponentModel/BackgroundWorker.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 namespace System.ComponentModel
 {
-    public class BackgroundWorker : IDisposable
+    public class BackgroundWorker : System.ComponentModel.Component
```
Comment (stephentoub) [inline src/System.ComponentModel.EventBasedAsync/src/System/ComponentModel/BackgroundWorker.cs]: IIRC, because Component wasn't a base class, we added finalizers to some of these derived types like BackgroundWorker, so that if types derived from BackgroundWorker and override Dispose(bool), they would still have their Dispose method called on finalization (since Component has one).  Now that we're adding back Component, we should remove such finalizers we added just to core.  In addition to being unnecessary, they could actually lead to problems, if they result in Dispose(false) being executed twice for an instance.

File: src/System.Diagnostics.Process/src/System/Diagnostics/ProcessModule.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@ namespace System.Diagnostics
     ///     a particular process.  Using this component, you can determine
     ///     information about the module.
     /// </devdoc>
-    public class ProcessModule
+    public class ProcessModule : System.ComponentModel.Component
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/src/System/Diagnostics/ProcessModule.cs]: For all of these, can we add ```using System.ComponentModel;``` rather than including the namespace here?

**Comment by stephentoub** [comment]
A few comments, otherwise LGTM

File: src/System.ComponentModel.EventBasedAsync/src/System/ComponentModel/BackgroundWorker.cs (hunk 1 above)
Comment (AlexGhiondea) [inline src/System.ComponentModel.EventBasedAsync/src/System/ComponentModel/BackgroundWorker.cs]: Indeed -- there were quite a few places where I had to remove the Dispose method we introduced before this.

File: src/System.Net.Ping/tests/FunctionalTests/System.Net.Ping.Functional.Tests.csproj [hunk 3]
```diff
@@ -34,7 +39,7 @@
       <Link>Common\tests\System\PlatformDetection.cs</Link>
     </Compile>
   </ItemGroup>
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)'==''">
     <ProjectReference Include="..\..\pkg\System.Net.Ping.pkgproj">
```
Comment (weshaggard) [inline src/System.Net.Ping/tests/FunctionalTests/System.Net.Ping.Functional.Tests.csproj]: This pkgproj ProjectReference should work for both 1.3 and 1.7 configurations. Why did you need to move to a package reference for some cases.

File: src/System.Net.Ping/tests/System.Net.Ping.Functional.Tests.builds [hunk 4]
```diff
@@ -8,6 +8,11 @@
     <Project Include="FunctionalTests\System.Net.Ping.Functional.Tests.csproj">
       <TestTFMs>netcoreapp1.0;net46</TestTFMs>
       <OSGroup>Windows_NT</OSGroup>
+      <TargetGroup>netstandard1.3</TargetGroup>
+    </Project>
+    <Project Include="FunctionalTests\System.Net.Ping.Functional.Tests.csproj">
```
Comment (weshaggard) [inline src/System.Net.Ping/tests/System.Net.Ping.Functional.Tests.builds]: Please put the default TargetGroup in the group at the top of the file see https://github.com/dotnet/corefx/pull/12312 for reference.

File: src/System.Net.Ping/tests/FunctionalTests/System.Net.Ping.Functional.Tests.csproj (hunk 3 above)
Comment (AlexGhiondea) [inline src/System.Net.Ping/tests/FunctionalTests/System.Net.Ping.Functional.Tests.csproj]: Without this the netcoreapp1.0 tests were not able to resolve the asset for it.

**Comment by stephentoub** [comment]
@AlexGhiondea, you didn't remove the finalizers as I mentioned?

**Comment by AlexGhiondea** [comment]
@stephentoub I was 100% sure you meant the Dispose methods we added. Created #12360 to remove the actual finalizers.
